com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/HooksProcessor.java

```diff
@@ -0,0 +1,70 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.util.List;
+
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.processor.EnvMtaMetadataParser;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.processor.MtaMetadataParser;
+import com.sap.cloud.lm.sl.cf.process.steps.AfterHooksPhaseProvider;
+import com.sap.cloud.lm.sl.cf.process.steps.BeforeHooksPhaseProvider;
+import com.sap.cloud.lm.sl.cf.process.steps.ProcessContext;
+import com.sap.cloud.lm.sl.cf.process.steps.StepPhase;
+import com.sap.cloud.lm.sl.mta.model.Hook;
+import com.sap.cloud.lm.sl.mta.model.Module;
+
+public class HooksProcessor {
+
+    private final ProcessContext context;
+    private final MtaMetadataParser mtaMetadataParser;
+    private final EnvMtaMetadataParser envMtaMetadataParser;
+
+    public HooksProcessor(ProcessContext context, MtaMetadataParser mtaMetadataParser, EnvMtaMetadataParser envMtaMetadataParser) {
+        this.context = context;
+        this.mtaMetadataParser = mtaMetadataParser;
+        this.envMtaMetadataParser = envMtaMetadataParser;
+    }
+
+    public boolean processBeforeHooks(BeforeHooksPhaseProvider beforeHooksPhaseProvider, StepPhase currentStepPhase) {
+        HooksHelper hooksHelper = getHooksHelperBuilder().withHookPhaseBeforeStep(beforeHooksPhaseProvider.getHookPhaseBeforeStep(context))
+                                                         .build();
+        Module moduleToDeploy = hooksHelper.determineModuleToDeploy(context);
+        return executeBeforeStepHooks(hooksHelper, currentStepPhase, moduleToDeploy);
+    }
+
+    public boolean processAfterHooks(AfterHooksPhaseProvider afterHooksPhaseProvider, StepPhase currentStepPhase) {
+        HooksHelper hooksHelper = getHooksHelperBuilder().withHookPhaseAfterStep(afterHooksPhaseProvider.getHookPhaseAfterStep(context))
+                                                         .build();
+        Module moduleToDeploy = hooksHelper.determineModuleToDeploy(context);
+        return executeAfterStepHooks(hooksHelper, currentStepPhase, moduleToDeploy);
+    }
+
+    private boolean executeBeforeStepHooks(HooksHelper hooksHelper, StepPhase currentStepPhase, Module moduleToDeploy) {
+        if (moduleToDeploy == null) {
+            return false;
+        }
+        if (!hooksHelper.isInPreExecuteStepPhase(currentStepPhase)) {
+            return false;
+        }
+        List<Hook> hooksForExecution = hooksHelper.getHooksForStepPhase(moduleToDeploy, currentStepPhase);
+        hooksHelper.setHooksForExecution(hooksForExecution);
+        return !hooksForExecution.isEmpty();
+    }
+
+    private boolean executeAfterStepHooks(HooksHelper hooksHelper, StepPhase currentStepPhase, Module moduleToDeploy) {
```

Comment:
This method and the one before it differ only in one `if` condition. Is it possible to refactor them to remove the duplication?